arch/risc-v/src/common/riscv_pmp.c

```diff
@@ -89,6 +88,32 @@ typedef struct pmp_entry_s pmp_entry_t;
  * Private Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: log2ceil
+ *
+ * Description:
+ *   Calculate the up-rounded power-of-two for input.
+ *
+ * Input Parameters:
+ *   x - Argument to calculate the power-of-two from.
+ *
+ * Returned Value:
+ *   Power-of-two for argument, rounded up.
+ *
+ ****************************************************************************/
+
+static uintptr_t log2ceil(uintptr_t x)
```

Comment:
should we extract to a common function

Reply:
Yes that could be useful, I just didn't know where to put it. Does a place for kernel internal / integer math functions exist ?

Reply:
put to include/nuttx/lib/math32.h too.